### PR TITLE
digestset: deprecate package in favor of go-digest/digestset

### PR DESCRIPTION
**Pull request by thaJeztah**
- follow-up to https://github.com/distribution/distribution/pull/3774
- moved it separate as need to look at the cncf-fuzzer project, as it's using this package for some of the fuzzing; see https://github.com/cncf/cncf-fuzzing/pull/144

---

This package was only used for the deprecated "shortid" syntax. Now that
support for this syntax was removed, we can also remove this package.

This patch deprecates and removes the package, adding temporary aliases pointing
to the new location to ease migration from docker/distribution to the new
distribution/distribution/v3. We should remove those aliases in a future update.

With these changes, the docs look like below:

Screenshot 2022-11-08 at 15 52 50

### CODE REVIEW

**Comment by thaJeztah** [comment]
Opened https://github.com/cncf/cncf-fuzzing/pull/236 as well to remove fuzzing for this package

**Comment by thaJeztah** [comment]
@milosgajdos @corhere PTAL

File: digestset/deprecated.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+import (
+	"github.com/opencontainers/go-digest"
+	"github.com/opencontainers/go-digest/digestset"
+)
+
+// ErrDigestNotFound is used when a matching digest
+// could not be found in a set.
+//
+// Deprecated: use [github.com/opencontainers/go-digest/digestset.ErrDigestNotFound].
```
Comment (corhere) [inline digestset/deprecated.go:11]: FYI: [the fully-qualified package path is not required for doc links to imported packages.](https://go.dev/doc/comment#links:~:text=When%20referring%20to,cannot%20be%20used.)
```suggestion
// Deprecated: use [digestset.ErrDigestNotFound].
```

File: digestset/deprecated.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+// ErrDigestAmbiguous is used when multiple digests
+// are found in a set. None of the matching digests
+// should be considered valid matches.
+//
+// Deprecated: use [github.com/opencontainers/go-digest/digestset.ErrDigestAmbiguous].
+var ErrDigestAmbiguous = digestset.ErrDigestAmbiguous
+
+// Set is used to hold a unique set of digests which
+// may be easily referenced by easily  referenced by a string
```
Comment (corhere) [inline digestset/deprecated.go:22]:
```suggestion
// may be easily referenced by a string
```

**Review by corhere — APPROVED** [review]

File: digestset/deprecated.go (hunk 1 above)
Comment (thaJeztah) [reply digestset/deprecated.go:11]: > FYI: [the fully-qualified package path is not required for doc links to imported packages.](https://go.dev/doc/comment#links:~:text=When%20referring%20to,cannot%20be%20used.)

Ah, yes, I actually knew that. I had that initially, but had to wiggle the code a bit to make the deprecations apply. Reason was that a _group of vars_ won't pick up the deprecation, so; something like below doesn't work;

```go
var (
    // Foo bla bla
    //
    // Deprecated: blablabla
    Foo string

    // Bar bla bla
    //
    // Deprecated: blablabla
    Bar string
)
```

I tried if changing some of the Deprecation line would work, then forgot to change it back 🙈 
 

Planning to open a bugreport for that in the Go issue tracker.

**Comment by thaJeztah** [comment]
> mind you if you accept the commits they won't add your signature and DCO will fail, so you wanna modify the code directly @thaJeztah

Thx for the reminder; yes the suggestions are mixed blessing; nice way to show "this is what I suggest", but horrible that they screw up your commits 😂 

Either way; updated; this one should be ready to go 👍

**Comment by thaJeztah** [comment]
Whoop, still green; @corhere ptal 👍

File: digestset/deprecated.go (hunk 2 above)
Comment (corhere) [reply digestset/deprecated.go:22]: Do you want to address this one @thaJeztah? This conversation was marked resolved without changing the code comment.

Comment (thaJeztah) [reply digestset/deprecated.go:22]: oh! overlooked it; mistook it for changing the documentation links; fixing now

Comment (thaJeztah) [reply digestset/deprecated.go]: updated; now it's done _for real_ 😂

**Review by corhere — APPROVED** [review]
The doc comment could be reflowed, but it's not a big deal as it'll look fine on pkg.go.dev.

**Comment by thaJeztah** [comment]
> The doc comment could be reflowed, but it's not a big deal as it'll look fine on pkg.go.dev.

Yeah, wanted to keep it close to the original for now (the aliases should also be temporarily 🤞)

**Pull request merged**